From d7c7030062c925a1a81a4aecb44197ba6ade4af3 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 12 Dec 2025 05:49:09 +0100
Subject: [PATCH] pinctrl: airoha: Add missing master mode bit for i2c1 func

The i2c1 function group require the i2c master mode bit to set to
correctly function.

Add the missing bit in addition to the 2nd i2c mode bit.

Fixes: 1c8ace2d0725 ("pinctrl: airoha: Add support for EN7581 SoC")
Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/pinctrl/mediatek/pinctrl-airoha.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/pinctrl/mediatek/pinctrl-airoha.c b/drivers/pinctrl/mediatek/pinctrl-airoha.c
index 547a798b71c8..648fe7700b32 100644
--- a/drivers/pinctrl/mediatek/pinctrl-airoha.c
+++ b/drivers/pinctrl/mediatek/pinctrl-airoha.c
@@ -820,7 +820,13 @@ static const struct airoha_pinctrl_func_group i2c_func_group[] = {
 			GPIO_2ND_I2C_MODE_MASK,
 			GPIO_2ND_I2C_MODE_MASK
 		},
-		.regmap_size = 1,
+		.regmap[1] = {
+			AIROHA_FUNC_MUX,
+			REG_GPIO_2ND_I2C_MODE,
+			GPIO_I2C_MASTER_MODE_MODE,
+			GPIO_I2C_MASTER_MODE_MODE
+		},
+		.regmap_size = 2,
 	},
 };
 
-- 
2.51.0

